Mathlib/GroupTheory/Subgroup/Basic.lean

```diff
@@ -0,0 +1,3736 @@
+
+@[to_additive]
+theorem comap_sup_comap_le (H K : Subgroup N) (f : G →* N) :
+    comap f H ⊔ comap f K ≤ comap f (H ⊔ K) :=
+  Monotone.le_map_sup (fun _ _ => comap_mono) H K
+#align subgroup.comap_sup_comap_le Subgroup.comap_sup_comap_le
+#align add_subgroup.comap_sup_comap_le AddSubgroup.comap_sup_comap_le
+
+@[to_additive]
+theorem supᵢ_comap_le {ι : Sort _} (f : G →* N) (s : ι → Subgroup N) :
+    (⨆ i, (s i).comap f) ≤ (supᵢ s).comap f :=
+  Monotone.le_map_supᵢ fun _ _ => comap_mono
+#align subgroup.supr_comap_le Subgroup.supᵢ_comap_le
+#align add_subgroup.supr_comap_le AddSubgroup.supᵢ_comap_le
+
+@[to_additive]
+theorem comap_inf (H K : Subgroup N) (f : G →* N) : (H ⊓ K).comap f = H.comap f ⊓ K.comap f :=
+  (gc_map_comap f).u_inf
+#align subgroup.comap_inf Subgroup.comap_inf
+#align add_subgroup.comap_inf AddSubgroup.comap_inf
+
+@[to_additive]
+theorem comap_infᵢ {ι : Sort _} (f : G →* N) (s : ι → Subgroup N) :
+    (infᵢ s).comap f = ⨅ i, (s i).comap f :=
+  (gc_map_comap f).u_infᵢ
+#align subgroup.comap_infi Subgroup.comap_infᵢ
+#align add_subgroup.comap_infi AddSubgroup.comap_infᵢ
+
+@[to_additive]
+theorem map_inf_le (H K : Subgroup G) (f : G →* N) : map f (H ⊓ K) ≤ map f H ⊓ map f K :=
+  le_inf (map_mono inf_le_left) (map_mono inf_le_right)
+#align subgroup.map_inf_le Subgroup.map_inf_le
+#align add_subgroup.map_inf_le AddSubgroup.map_inf_le
+
+@[to_additive]
+theorem map_inf_eq (H K : Subgroup G) (f : G →* N) (hf : Function.Injective f) :
+    map f (H ⊓ K) = map f H ⊓ map f K := by
+  rw [← SetLike.coe_set_eq]
+  simp [Set.image_inter hf]
+#align subgroup.map_inf_eq Subgroup.map_inf_eq
+#align add_subgroup.map_inf_eq AddSubgroup.map_inf_eq
+
+@[to_additive (attr := simp)]
+theorem map_bot (f : G →* N) : (⊥ : Subgroup G).map f = ⊥ :=
+  (gc_map_comap f).l_bot
+#align subgroup.map_bot Subgroup.map_bot
+#align add_subgroup.map_bot AddSubgroup.map_bot
+
+@[to_additive (attr := simp)]
+theorem map_top_of_surjective (f : G →* N) (h : Function.Surjective f) : Subgroup.map f ⊤ = ⊤ := by
+  rw [eq_top_iff]
+  intro x _
+  obtain ⟨y, hy⟩ := h x
+  exact ⟨y, trivial, hy⟩
+#align subgroup.map_top_of_surjective Subgroup.map_top_of_surjective
+#align add_subgroup.map_top_of_surjective AddSubgroup.map_top_of_surjective
+
+@[to_additive (attr := simp)]
+theorem comap_top (f : G →* N) : (⊤ : Subgroup N).comap f = ⊤ :=
+  (gc_map_comap f).u_top
+#align subgroup.comap_top Subgroup.comap_top
+#align add_subgroup.comap_top AddSubgroup.comap_top
+
+/-- For any subgroups `H` and `K`, view `H ⊓ K` as a subgroup of `K`. -/
+@[to_additive "For any subgroups `H` and `K`, view `H ⊓ K` as a subgroup of `K`."]
+def subgroupOf (H K : Subgroup G) : Subgroup K :=
+  H.comap K.subtype
+#align subgroup.subgroup_of Subgroup.subgroupOf
+#align add_subgroup.add_subgroup_of AddSubgroup.addSubgroupOf
+
+/-- If `H ≤ K`, then `H` as a subgroup of `K` is isomorphic to `H`. -/
+@[to_additive "If `H ≤ K`, then `H` as a subgroup of `K` is isomorphic to `H`.", simps]
```

Comment:
Does this transfer the `simps` as well, or does it not need to?

Reply:
no, this needs fixing